<?xml version="1.0" ?>
<launch>

  <arg name="cpp" default="true"/>
  <arg name="bluefox" default="false"/>
  <arg name="flea3" default="true"/>
  <arg name="openni" default="false"/>
  <arg name="image_topic" default="/pouring_control/pg_13344889/image_raw" />



  <!-- Desired height in cm -->
  <arg name="des_height_cm" default="2.4"/>

  <!--options: L,R,M,A -->
  <!-- Left now, as robot pours from the left, hence M,R are in jeporady -->
  <arg name="vis_side" default="-selection L"/>

  <!-- Runs the camera drivers -->
<!--   <include file="$(find queen_cam_calibration)/launch/kinect_calib.launch">
    <arg name="bluefox" value="$(arg bluefox)"/>
    <arg name="flea3" value="$(arg flea3)"/>
  </include>
 -->
  <!-- Runs the Apriltag node -->
  <!--<include file="$(find liquid_apriltags)/launch/new_cylinder_tag_detection.launch">
    <arg name="image" value="$(arg image_topic)"/>
  </include>-->
  <include file="$(find apriltag_ros)/launch/pose_estimator.launch">
    <arg name="camera" value="/pouring_control/pg_13344889"/>
    <arg name="border_size" value="1"/>
  </include>

  <node name='Level_Detector_Node_cpp' pkg='liquid_level_detection_nn' type='level_detect_node_nn' output="screen" if="$(arg cpp)">
    <param name="desired_height_cm" type="double" value="$(arg des_height_cm)"/>
    <param name="apriltag_cut_area" type="double" value="0.025"/>
    <param name="apriltag_size" type="double" value="0.019"/>
    <param name="cylinder_height" type="double" value="0.089"/>
    <param name="x_offset_pix" type="int" value="200"/>
    <param name="y_offset_pix" type="int" value="80"/>
    <param name="background_sub_history" type="int" value="250"/>
    <param name="background_sub_threshold" type="int" value="16"/>
    <param name="debug" type="bool" value="false"/>
    <remap from="~display_image" to="$(arg image_topic)"/>
    <remap from="~image" to="/hed_level_detector/output"/>
    <remap from="~detected_height" to="LDHeight"/>
    <remap from="apriltags" to="/pouring_control/apriltags"/>
    <remap from="~LDOutput" to="LDOutput"/>
    <remap from="~LDOutput_fg" to="LDOutput_fg"/>
  </node>

  <!--height for glass cylinder 0.089m tagsize: 0.017, yoffset was 40 but that was too low, area 41.6666-->
  <!--Front tag0 was 0.02m and height of 0.091 in front -->
  <!-- New plastic reciever height is 7.6cm, tagsize = 0.019cm, Area: 38.515cm**2 -->
  <node name="docker_wrapper" pkg="liquid_level_detection_nn" type="detector.py" respawn="true" output="screen">
    <remap from="/pg_13344889/image_raw" to="$(arg image_topic)"/>
    <remap from="/pg_13344889/tags" to="/pouring_control/pg_13344889/tags"/>
  </node>

</launch>
